### ISSUE TITLE
Using ElasticSearch mapper-attachment

### ISSUE THREAD

**Issue by frague59** [torchbox/wagtail#3200]
Hi, 

I'm trying to use the mapper-attachment with an instance of ES 1.7 with mapper-attachment v 2.7, and I've a serialization error when trying to update the index.

### What I've done

I've used a customized version of Abstract document:

```python
class OfficeDocument(AbstractDocument):
    """
    An indexable document with ``attachment``,
    using `elasticsearch-mapper-attachments <https://github.com/elastic/elasticsearch-mapper-attachments>`_
    """
    admin_form_fields = (
        'title',
        'file',
        'collection',
        'tags'
    )
    search_fields = CollectionMember.search_fields + [
        index.SearchField('title', partial_match=True, boost=10),
        index.SearchField('file', search_content=True),
        index.RelatedFields('tags', [
            index.SearchField('name', partial_match=True, boost=10),
        ]),
        index.FilterField('uploaded_by_user'),
    ]
```

And I have set the: 
```python
    WAGTAILDOCS_DOCUMENT_MODEL = "publications.OfficeDocument"
```

Then, I've linked this doc into a page: 
```python 
class OfficeDocumentPage(PublicationItemPage):
    """
    Page displaying an office document
    """
    document = models.ForeignKey(settings.WAGTAILDOCS_DOCUMENT_MODEL,
                                 related_name='+',
                                 on_delete=models.SET_NULL,
                                 null=True)

    class Meta:
        verbose_name = _('Office document')
        verbose_name_plural = _('Office documents')

    def __str__(self):
        return u'%s (%s)' % (self.title, _('Office document'))

... 
```

When I try to index the documents with:
```bash
$ ./manage.py update_index
Updating backend: default
default: Rebuilding index intranet-dev
default: home.HomePage             
default: publications.PlaceHolderPage .
default: publications.EventPage    .
default: publications.NewsTopicPage 
default: publications.OfficeDocument Traceback (most recent call last):
  File "./manage.py", line 12, in <module>
    execute_from_command_line(sys.argv)
  File "/home/fguerin/Boulot/workspace/projects/intranet.ville.tg/virtualenv/local/lib/python2.7/site-packages/django/core/management/__init__.py", line 367, in execute_from_command_line
    utility.execute()
  File "/home/fguerin/Boulot/workspace/projects/intranet.ville.tg/virtualenv/local/lib/python2.7/site-packages/django/core/management/__init__.py", line 359, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/home/fguerin/Boulot/workspace/projects/intranet.ville.tg/virtualenv/local/lib/python2.7/site-packages/django/core/management/base.py", line 294, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/home/fguerin/Boulot/workspace/projects/intranet.ville.tg/virtualenv/local/lib/python2.7/site-packages/django/core/management/base.py", line 345, in execute
    output = self.handle(*args, **options)
  File "/home/fguerin/Boulot/workspace/projects/intranet.ville.tg/virtualenv/local/lib/python2.7/site-packages/wagtail/wagtailsearch/management/commands/update_index.py", line 120, in handle
    self.update_backend(backend_name, schema_only=options['schema_only'])
  File "/home/fguerin/Boulot/workspace/projects/intranet.ville.tg/virtualenv/local/lib/python2.7/site-packages/wagtail/wagtailsearch/management/commands/update_index.py", line 87, in update_backend
    index.add_items(model, chunk)
  File "/home/fguerin/Boulot/workspace/projects/intranet.ville.tg/virtualenv/local/lib/python2.7/site-packages/wagtail/wagtailsearch/backends/elasticsearch.py", line 550, in add_items
    bulk(self.es, actions)
  File "/home/fguerin/Boulot/workspace/projects/intranet.ville.tg/virtualenv/local/lib/python2.7/site-packages/elasticsearch/helpers/__init__.py", line 190, in bulk
    for ok, item in streaming_bulk(client, actions, **kwargs):
  File "/home/fguerin/Boulot/workspace/projects/intranet.ville.tg/virtualenv/local/lib/python2.7/site-packages/elasticsearch/helpers/__init__.py", line 161, in streaming_bulk
    for bulk_actions in _chunk_actions(actions, chunk_size, max_chunk_bytes, client.transport.serializer):
  File "/home/fguerin/Boulot/workspace/projects/intranet.ville.tg/virtualenv/local/lib/python2.7/site-packages/elasticsearch/helpers/__init__.py", line 60, in _chunk_actions
    data = serializer.dumps(data)
  File "/home/fguerin/Boulot/workspace/projects/intranet.ville.tg/virtualenv/local/lib/python2.7/site-packages/elasticsearch/serializer.py", line 50, in dumps
    raise SerializationError(data, e)
elasticsearch.exceptions.SerializationError: ({'content_type': u'wagtaildocs_abstractdocument_publications_officedocument', 'title': u'Annuaire des \xe9lus', u'collection_id_filter': 1L, u'uploaded_by_user_id_filter': 1L, 'file': <FieldFile: documents/Annuaire_des_élus_sept_2014.pdf>, u'_partials': [u'Annuaire des \xe9lus'], 'pk': '1', 'tags': []}, TypeError("Unable to serialize <FieldFile: documents/Annuaire_des_\xc3\xa9lus_sept_2014.pdf> (type: <class 'django.db.models.fields.files.FieldFile'>)",))
```
### Technical details

* python --version : Python 2.7.12+
* Django version: 1.10.4
* Wagtail version: 1.7

**Comment by gasman** [comment]
I believe this is being addressed by #3028. (@kaedroho, can you confirm?)

**Comment by frague59** [comment]
I've got it, I'm too quick :)
I'll put some comments in my code, and wait for 1.8 release...
